### PR TITLE
Moved /src/Http/Features/ to /src/Extensions/Features/

### PR DESCRIPTION
**Pull request by vstr**
- [x] You've read the [Contributor Guide](https://github.com/dotnet/aspnetcore/blob/main/CONTRIBUTING.md) and [Code of Conduct](https://github.com/dotnet/aspnetcore/blob/main/CODE-OF-CONDUCT.md).
- [ ] You've included unit or integration tests for your change, where applicable.
- [ ] You've included inline docs for your change, where applicable.
- [x] There's an open issue for the PR that you are making. If you'd like to propose a new feature or change, please open an issue to discuss the change or find an existing issue.

**PR Title**
Moved /src/Http/Features/ to /src/Extensions/Features/

**PR Description**
Moved Microsoft.Extensions.Features projects to new location, updated references for sln, slnf and props.

Addresses #34423

### CODE REVIEW

**Review by Tratcher — APPROVED** [review]
Very thorough, thanks!

**Comment by Tratcher** [comment]
Looks like this needs one more step to fix the Code Check failures.

https://github.com/dotnet/aspnetcore/blob/main/docs/ReferenceResolution.md
> Run eng/scripts/GenerateProjectList.ps1 (or build.cmd /t:GenerateProjectList) when adding new projects

Can you give that a try?

**Comment by vstr** [comment]
Yes, trying to fix the build for few days, `GenerateProjectList.ps1` removes `Microsoft.Extensions.Features` from `eng\ProjectReferences.props` and `eng\SharedFramework.Local.props` for some reason. After that the build obviously fails locally.

The log message from 'azure-pipelines / aspnetcore-ci' says 
_"eng\ProjectReferences.props(,): error : Generated code is not up to date in eng/ProjectReferences.props. You might need to regenerate the reference assemblies or project list (see docs/ReferenceAssemblies.md and docs/ReferenceResolution.md)"_

From `docs/ReferenceResolution.md`:
- GenerateProjectList.ps1 removes `Microsoft.Extensions.Features`
- BaseLineGenerator - we don't have 'Microsoft.Extensions.Features' in `Baseline.xml` so there seems no need to run it.

There is no such doc as `docs/ReferenceAssemblies.md` (looks like the log message needs to be updated)

Any ideas on what may be missing here?

**Comment by dougbu** [comment]
> Any ideas on what may be missing here?

You need to update the two `Include` lists in https://github.com/dotnet/aspnetcore/blob/6bf34a09951ef0752c856b34a2cef85de164cd48/eng/Build.props#L155-L240 to include `$(RepoRoot)src\Extensions\**\*.csproj;` and `$(RepoRoot)src\WebEncoders\**\src\*.csproj;`, respectively.

**Comment by vstr** [comment]
Thank you @dougbu! it now builds without issues locally.

File: eng/Build.props [hunk 1]
```diff
@@ -192,6 +192,8 @@
                           $(RepoRoot)src\ProjectTemplates\*\*.csproj;
                           $(RepoRoot)src\submodules\spa-templates\src\*.csproj;
                           $(RepoRoot)src\ProjectTemplates\testassets\*\*.csproj;
+                          $(RepoRoot)src\Extensions\**\*.csproj;
+                          $(RepoRoot)src\WebEncoders\**\src\*.csproj;
```
Comment (dougbu) [inline eng/Build.props:196]:
```suggestion
                          $(RepoRoot)src\Extensions\**\*.csproj;
```

File: eng/Build.props [hunk 2]
```diff
@@ -230,6 +232,8 @@
                           $(RepoRoot)src\WebEncoders\**\src\*.csproj;
                           $(RepoRoot)src\HealthChecks\**\src\*.csproj;
                           $(RepoRoot)src\Testing\**\src\*.csproj;
+                          $(RepoRoot)src\Extensions\**\*.csproj;
+                          $(RepoRoot)src\WebEncoders\**\src\*.csproj;
```
Comment (dougbu) [inline eng/Build.props:236]:
```suggestion
                          $(RepoRoot)src\Extensions\**\src\*.csproj;
```

**Review by dougbu — CHANGES_REQUESTED** [review]

File: eng/Build.props [hunk 3]
```diff
@@ -192,6 +192,7 @@
                           $(RepoRoot)src\ProjectTemplates\*\*.csproj;
                           $(RepoRoot)src\submodules\spa-templates\src\*.csproj;
                           $(RepoRoot)src\ProjectTemplates\testassets\*\*.csproj;
+                          $(RepoRoot)src\Extensions\**\*.csproj
```
Comment (dougbu) [inline eng/Build.props:195]:
```suggestion
                          $(RepoRoot)src\Extensions\**\*.csproj;
```

**Review by dougbu — APPROVED** [review]
Sorry for the extra commits, something went wrong in GitHub UI and it didn't let me just delete the extra lines.

**Comment by vstr** [comment]
Thank you for your input anyway!

**Comment by dougbu** [comment]
Still need to run the update script @vstr

**Comment by vstr** [comment]
Finally all checks passed, thank you @dougbu , @Tratcher !

**Pull request merged**